tests/test_roc.py

```diff
@@ -161,3 +175,199 @@ def test_fpr_and_tpr_different_types(fpr, tpr):
+def test_roc_multi_from_raw_data(roc_multi_classification_raw_data):
+    y_test, y_score = roc_multi_classification_raw_data
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc_multi_with_labels"])
+def test_roc_multi_from_raw_data_with_labels(roc_multi_classification_raw_data_set2):
+    y_test, y_score = roc_multi_classification_raw_data_set2
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc_multi", "roc_multi"])
+def test_roc_multi(roc_multi_classification_values):
+    fpr, tpr, label = roc_multi_classification_values
+    # test when fpr and tpr are lists
+    multi_roc = plot.ROC(fpr, tpr, label=label)
+    multi_roc.plot()
+
+    # test when fpr and tpr are np.arrays
+    fpr_np_array = [np.array(fpr_list) for fpr_list in fpr]
+    tpr_np_array = [np.array(tpr_list) for tpr_list in tpr]
+    multi_roc = plot.ROC(fpr_np_array, tpr_np_array, label=label)
+    multi_roc.plot()
+
+
+@image_comparison(baseline_images=["roc", "roc2", "roc_add_roc"])
+def test_roc_add_to_roc_from_raw_data(y):
+    y_test, y_score = y
+    y_test_roc2 = np.array([1, 1, 1, 1, 1, 0, 0, 0, 0, 0])
+    y_score_roc2 = np.array(
+        [
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            [0.3, 0.6, 0.7, 0.8, 0.9, 0.9, 0.8, 0.2, 0.3, 0.5],
+        ]
+    ).T
+
+    roc1 = plot.ROC.from_raw_data(y_test, y_score)
+    roc2 = plot.ROC.from_raw_data(y_test_roc2, y_score_roc2)
+    roc1 + roc2
+
+
+@image_comparison(baseline_images=["roc_add_roc"])
+def test_roc_add_to_roc(roc_values):
+    fpr1, tpr1 = roc_values
+
+    fpr2 = [0.0, 0.2, 0.4, 0.4, 0.6, 0.8, 1.0]
+    tpr2 = [0.0, 0.2, 0.4, 0.8, 0.8, 1.0, 1.0]
+
+    roc1 = plot.ROC(fpr1, tpr1)
+    roc2 = plot.ROC(fpr2, tpr2)
+
+    roc1 + roc2
+
+
+@image_comparison(baseline_images=["roc_add_multi"])
+def test_roc_add_to_multi(roc_values, roc_multi_classification_values):
+    fpr1, tpr1 = roc_values
+    fpr2, tpr2, label = roc_multi_classification_values
+
+    roc1 = plot.ROC(fpr1, tpr1)
+    roc2 = plot.ROC(fpr2, tpr2, label=label)
+    roc1 + roc2
+
+
+@image_comparison(baseline_images=["multi_add_roc"])
+def test_roc_multi_add_to_roc(roc_values, roc_multi_classification_values):
+    fpr1, tpr1, label = roc_multi_classification_values
+
+    fpr2, tpr2 = roc_values
+
+    roc1 = plot.ROC(fpr1, tpr1, label=label)
+    roc2 = plot.ROC(fpr2, tpr2)
+    roc1 + roc2
+
+
+@image_comparison(
+    baseline_images=["roc_multi", "roc_multi_set2", "roc_multi_add_multi"]
+)
+def test_roc_multi_add_to_multi_from_raw_data(
+    roc_multi_classification_raw_data, roc_multi_classification_raw_data_set2
+):
+    y_test, y_score = roc_multi_classification_raw_data
+    y_test_roc2, y_score_roc2 = roc_multi_classification_raw_data_set2
+
+    roc1 = plot.ROC.from_raw_data(y_test, y_score)
+    roc2 = plot.ROC.from_raw_data(y_test_roc2, y_score_roc2)
+    roc1 + roc2
+
+
+@image_comparison(baseline_images=["roc_multi_set2", "roc_multi_add_multi"])
```

Comment:
One more image can be added to the list if the plot is generating in the test (roc1, roc2, roc1+roc2)